### DISCUSSION TITLE
Trying to set up useful layers and to understand basic functions on the linguistic program INCEpTION

### DISCUSSION THREAD

**Discussion by engelbrechtdesign** [inception-project/inception#4997] [Q&A]
https://github.com/user-attachments/assets/61015c54-ba87-4cd7-9d90-dfb2c34f3d6c

So far, I have understood that as a user, I can manually assign various annotations, even such as simple numerical values from -2 to +2, to any word or phrase in a text. I also understand that the program remembers the words marked with such ratings/annotations and autonomously marks more of them in a meaningful way in the subsequent text, suggesting similar (or slightly different) ratings. By confirming or correcting these suggestions, the program "learns" by time to make these suggestions more accurately.

1. Question: Does this learning process only apply to the project in which it is conducted, or can it be transferred to any newly created project by inserting a layer with the same name and further developing it there?  In other words, do I have to store all the different texts to be annotated in this way in the same project, or can I organize them into different projects by retaining the acquired advantages?
2. Can the program also automatically highlight for examples all VERBS or all ADVERBS within a text in color by ITSELF, so that it is easier for the user to find them in order to manually annotate each of them with a value? How can it be told to do this?

**Comment by reckart** [comment]
> Question: Does this learning process only apply to the project in which it is conducted, or can it be transferred to any newly created project by inserting a layer with the same name and further developing it there? In other words, do I have to store all the different texts to be annotated in this way in the same project, or can I organize them into different projects by retaining the acquired advantages?

Those recommenders that learn do so only from the current user's annotation. So if you'd access the project as another user who has made different annotations, you'd get different suggestions from the recommenders. 

Currently only the data of one recommender can be exported. If you use a String Matcher recommender, you can open the recommender sidebar on the annotation page and there is a button to export the model below the recommender status info box.
This model is actually a **gazeteer** and you can import that again on the settings panel of another String Matcher recommender.

For other recommenders, model export (or import) is currently not possible. But I should probably have a look at that again... might be time by now to add such a feature...

> Can the program also automatically highlight for examples all VERBS or all ADVERBS within a text in color by ITSELF, so that it is easier for the user to find them in order to manually annotate each of them with a value? How can it be told to do this?

There is no built-in part-of-speech tagger in INCEpTION. A person familiar with Python programming could hook one up as an external recommender though. We have a [spacy POS classifier example](https://github.com/inception-project/inception-external-recommender/blob/main/ariadne/contrib/spacy.py#L57-L80) in our [external recommender repository](https://github.com/inception-project/inception-external-recommender).
